src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/Models/RazorCodeAction.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using MediatR;
+using Newtonsoft.Json;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Models
+{
+
+    [JsonObject]
+    [DebuggerDisplay("{Title,nq}")]
+    internal class RazorCodeAction : CodeAction, IRequest<RazorCodeAction>, IBaseRequest
+    {
+        // RazorCodeAction inherits from CodeAction to ensure we can utilize additional properties
+        // Re-implements the properties to ensure we have direct control over property (de)serialization. 
+        // Extensive platform compatibility issues occur without this.
+
+        [JsonProperty(PropertyName = "title", Required = Required.Always)]
+        public new string Title { get; set; }
```

Comment:
So the problem with using new on all of these if we have any types that reference the original CodeAction type it's going to get the incorrect value. Is that ok? I guess the real question is do we imagine doing all of this will be worth it from a serialization/deserialization perf perspective?

Reply:
Wouldn't the `new` types simply supersede the existing base properties? 

I'm not _too_ worried about perf/serialization ramifications as this is almost a direct copy of what we [had before](https://github.com/dotnet/aspnetcore-tooling/pull/2669/files) but removed as it was no longer needed (we were using it to add the `CodeAction.Data` property which subsequently got added to O#.CodeAction).

Reply:
Is there a recommended alternative to using the new property? I remember having pretty extensive issues had I just left it out entirely.

Reply:
So the `new` property may not do what you think. For instance consider the following:
```C#
var codeAction = new RazorCodeAction()
{
    Title = "John",
};

var castedCodeAction = codeAction as CodeAction;
castedCodeAction.Title = "Bob";

Console.WriteLine(codeAction);
Console.WriteLine(castedCodeAction);
```

You'd get:
```
John
Bob
```

Reply:
I'd be curious as to why leaving it off wouldn't work other than to better control serialization/deserialization

Reply:
Thanks for the explanation, appreciate it 😄 

Hmm, so it's working fine without those new properties, not sure what the issue was before (perhaps VSCode when I try to integrate with that?).

Reply:
> Hmm, so it's working fine without those new properties, not sure what the issue was before (perhaps VSCode when I try to integrate with that?).

VSCode would barf if the serialized output was incompatible 😄